packages/core/src/utils/index.ts

```diff
@@ -138,3 +138,9 @@ export function unwrapInputListParam<Types extends SchemaTypes>(
 
   return param;
 }
+
+export function isPromiseLike<T>(value: unknown): value is PromiseLike<T> {
```

Comment:
I think there is an existing helper (isThenable) that does the same thing

Reply:
Updated to use that, and changed `isThenable` to return `PromiseLike`